packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -71,6 +73,35 @@ const extractAttachedFilesMetadata = (metadata: Record<string, string>): FileMet
   }
 };
 
+/* @conditional-compile-remove(teams-inline-images) */
+const extractInlineImageFilesMetadata = (attachments: ChatAttachment[]): FileMetadata[] => {
+  return attachments.map((attachment) => ({
+    attachmentType: attachment.attachmentType,
+    id: attachment.id,
+    name: attachment.name ?? '',
+    extension: attachment.contentType ?? '',
+    url: attachment.url,
+    previewUrl: attachment.previewUrl
+  }));
+};
+
+/* @conditional-compile-remove(file-sharing) @conditional-compile-remove(teams-inline-images) */
+const extractFilesMetadata = (message: ChatMessageWithStatus): FileMetadata[] => {
```

Comment:
I thought e2e is like e2e test :D, it's like get it working end 2 end

Reply:
LOL yes. This PR was gluing the adapter and the composite pieces together